libs/components/ag-grid/src/lib/modules/ag-grid/ag-grid-wrapper.component.ts

```diff
@@ -287,34 +287,41 @@ export class SkyAgGridWrapperComponent
     }
   }
 
-  #updateGridTheme(themeSettings?: SkyThemeSettings): void {
-    let agTheme: 'default' | 'modern-light' | 'modern-dark';
-    if (themeSettings?.theme.name === 'modern') {
-      agTheme = `modern-${themeSettings.mode.name}` as
-        | 'modern-light'
-        | 'modern-dark';
-    } else {
-      agTheme = `default`;
-    }
-    const variation = this.#hasEditableClass ? 'data-entry-grid' : 'data-grid';
+  #updateGridTheme(
+    hasEditableClass: boolean,
+    themeSettings?: SkyThemeSettings,
+  ): void {
+    const agTheme = agGridTheme(hasEditableClass, themeSettings, this.compact);
+    const previousValue = this.#wrapperClasses.getValue();
+    const previousTheme = previousValue.find((c) => c.startsWith('ag-theme-'));
     let value = [
-      ...this.#wrapperClasses
-        .getValue()
-        .filter((c) => !c.startsWith('ag-theme-')),
-      `ag-theme-sky-${variation}-${agTheme}`,
+      ...previousValue.filter((c) => !c.startsWith('ag-theme-')),
+      agTheme,
     ];
     const textSelectionClass = 'sky-ag-grid-text-selection';
-    if (this.#getTextSelection()) {
+    if (this.#getTextSelection(hasEditableClass)) {
       value.push(textSelectionClass);
     } else {
       value = value.filter((c) => c !== textSelectionClass);
     }
     this.#wrapperClasses.next([...new Set(value)]);
+    if (this.agGrid?.api && !this.agGrid.api.isDestroyed()) {
+      if (this.agGrid?.api?.getGridOption('domLayout') !== 'autoHeight') {
+        this.agGrid?.api?.resetRowHeights();
+      }
+      this.agGrid?.api?.refreshHeader();
+      if (
+        previousTheme?.endsWith('-compact') !==
```

Comment:
might be nice to make this a function similar to `agGridThemeIsCompact`, maybe `agGridHasCompactClass`?